frappe/website/website_generator.py

```diff
@@ -177,4 +177,4 @@ def update_website_search_index(self):
 			frappe.enqueue(update_index_for_path, path=self.route)
 		elif self.route:
 			# If the website is not published
-			remove_document_from_index(self.route)
+			frappe.enqueue(remove_document_from_index, path=self.route)
```

Comment:
```suggestion
			frappe.enqueue(remove_document_from_index, path=self.route, enqueue_after_commit=True)
```

Reply:
If the job starts before you commit, it will see old data from DB and do nothing.